daemon/logger/copier_test.go

```diff
@@ -102,6 +106,50 @@ func TestCopier(t *testing.T) {
 	}
 }
 
+// TestCopierTimestamp tests that Copier is using a time function to add timestamp to a message
+func TestCopierTimestamp(t *testing.T) {
+	timeFunc := func() time.Time {
+		return time.Date(2017, time.June, 0, 0, 0, 0, 0, time.UTC)
+	}
+	var stdout bytes.Buffer
+	if _, err := stdout.WriteString("hello world\n"); err != nil {
+		t.Fatal(err)
+	}
+	var jsonBuf bytes.Buffer
+
+	jsonLog := &TestLoggerJSON{Encoder: json.NewEncoder(&jsonBuf)}
+
+	c := NewCopier(
+		map[string]io.Reader{
+			"stdout": &stdout,
+		},
+		jsonLog, timeFunc)
+	c.Run()
+	wait := make(chan struct{})
+	go func() {
+		c.Wait()
+		close(wait)
+	}()
+	select {
+	case <-time.After(1 * time.Second):
+		t.Fatal("Copier failed to do its work in 1 second")
```

Comment:
Is this goroutine and select just to set a timeout in the case of an error?

Go test already has a timeout for the test run which might be sufficient?

If not, maybe a helper like `TestMustRunIn(time.Duration)` would make this more obvious? I think it could just panic in a goroutine which will fail the test, and help this test be more readable.

Reply:
TBH, I did copy-paste on one of the tests in the file and force it to test my change.
Thanks for your review. I will go over this again 👍